x-pack/plugins/alerting/server/application/rule/methods/aggregate/aggregate.ts

```diff
@@ -6,19 +6,15 @@
  */
 
 import { KueryNode, nodeBuilder } from '@kbn/es-query';
-import type { AggregationsAggregationContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
-import { AlertingAuthorizationEntity } from '../../authorization';
-import { ruleAuditEvent, RuleAuditAction } from '../common/audit_events';
-import { buildKueryNodeFilter } from '../common';
-import { alertingAuthorizationFilterOpts } from '../common/constants';
-import { RulesClientContext } from '../types';
-import { RawRule, AggregateOptions } from '../../types';
-import { validateRuleAggregationFields } from '../lib/validate_rule_aggregation_fields';
+import { RuleAttributes } from '../../../../data/rule/types';
+import { AlertingAuthorizationEntity } from '../../../../authorization';
+import { ruleAuditEvent, RuleAuditAction } from '../../../../rules_client/common/audit_events';
+import { buildKueryNodeFilter } from '../../../../rules_client/common';
+import { alertingAuthorizationFilterOpts } from '../../../../rules_client/common/constants';
+import { RulesClientContext } from '../../../../rules_client/types';
 
-export interface AggregateParams<AggregationResult> {
-  options?: AggregateOptions;
-  aggs: Record<keyof AggregationResult, AggregationsAggregationContainer>;
-}
+import type { AggregateParams } from './types';
```

Comment:
I think something we're trying to do with the HTTP versioning effort is to implement strict input validation for all of our `rulesClient` methods (since they're shared by other plugins as well).

This means deriving the params types from a schema and validating the schema in the method (see `create_rule.ts` for examples of this). We could potentially also validate the output schema too.

Reply:
I thought we talked about ignoring this one as there was some issue with it. We do the validation with `validateRuleAggregationFields` which I versioned under `application/rule/methods/aggregate/validation/validate_rule_aggregation_fields/v1.ts`

Reply:
right but that only validates the aggregation query? Would be nice to validate the options too

Reply:
Done (I think) https://github.com/elastic/kibana/pull/164284/commits/cb1bc9adf3aa897e297ed7edbee384994a579853